### ISSUE TITLE
`SecurityIdentityAugmentor` are called multiple times with multiple identityProviders

### ISSUE THREAD

**Issue by Malandril** [quarkusio/quarkus#42600]
### Describe the bug

With quarkus security, when multiple `IdentityProvider` are present and they return a `nullitem`  the SecurityIdentityAugmentors will be called once per IdentityProvider  that where called.

### Expected behavior

If an identity provider returns a null item, it should not re trigger an augmentation. The augmentor should be called only once.

### How to Reproduce?

[https://github.com/Malandril/quarkus-bug-augment](https://github.com/Malandril/quarkus-bug-augment)

### Additional information

 #42599 with fix

**Comment by sberyozkin** [comment]
I don't think it is a valid issue. The augmentor may return null for whatever the first identity provider has produced but decide to augment the identity the next time.
What exactly is the problem with it called multiple times in this case ?

**Comment by sberyozkin** [comment]
@Malandril If you really don't want to the augmentor multiple times then you may want to use a new augmentor method accepting the routing context map to coordinate. 
Or if possible do the conditional augmentation based on the passed identity type. 

I don't think we can block additional identity providers from having their identities checked by augmentors based on what happened with the previous identity provider

CC @michalvavrik

**Comment by Malandril** [comment]
This happens when the IdentityProvider return `Uni.createfrom().nullitem()` i don't see why it should change the number if times the augmentors are called.

This Can be problematic if the Augmentor calls an expensive operation, then it will be called multiple times with the "same identity" doing work, to add roles to an identity which was already augmented.

If this is expected behaviour then it should be documented so that when crating custom Augmentors we need to handle cases where the "same identity" may be passed multiple times

**Comment by sberyozkin** [comment]
@Malandril Oh, sorry,  you mean the identity provider retuning null...
Then your PR looks valid,  though we'll have a look,  next week, thanks

**Comment by michalvavrik** [comment]
> If this is expected behaviour then it should be documented so that when crating custom Augmentors we need to handle cases where the "same identity" may be passed multiple times

It's just an oversight. Night catch, I think it is there almost from initial augmentor version and noone mentioned :-)

**Comment by michalvavrik** [comment]
ad expected behavior

> If an identity provider returns a null item, it should not re trigger an augmentation. Each augmentor should be called only once per identity.

I think this is confusing because what you PR does is not handling returns of null, but instead repeated application of augmentors. What should happen when `identity provider returns null item` is throwing authentication exception as we do for single provider.

**Comment by sberyozkin** [comment]
@michalvavrik Why should an authentication exception be thrown if, in the presence of 3 providers, one returns null ?

**Comment by sberyozkin** [comment]
Never mind, I seem to have problems completing reading the whole comment carefully recently...

**Comment by michalvavrik** [comment]
> Never mind, I seem to have problems completing reading the whole comment carefully recently...

np, I think it already happens https://github.com/quarkusio/quarkus/blob/6b0a25dee37fa205cb0351fd45577dbc9462bef9/extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusIdentityProviderManagerImpl.java#L117 but it took me while to figure what this issue is about and "expected behavior" confused me.
